app/main/fund.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import re
+from dataclasses import dataclass
+from enum import StrEnum
+
+from config import Config
+
+
+class SupportedRole(StrEnum):
+    TF_SUBMITTER_ROLE = "TF_MONITORING_RETURN_SUBMITTER"
+
+
+class SupportedFund(StrEnum):
+    TOWNS_FUND = "Towns Fund"
+
+
+@dataclass
+class FundConfig:
+    fund_name: SupportedFund
+    reporting_period: str
+    reporting_round: int
+    deadline: datetime.date
+    email: str
+
+    def __init__(
+        self, fund_name: SupportedFund, reporting_period: str, reporting_round: int, deadline: datetime.date, email: str
+    ):
+        """Initialises a FundConfig.
+
+        Applies input validation to prevent downstream errors. This is to mitigate against the dangers of storing the
+        FundConfig state in code, where it can be easily changed.
+
+        :param fund_name:  The name of the fund. Must be a string.
+        :param reporting_period: The reporting period. Must be a string.
+        :param reporting_round: The reporting round. Must be an int.
+        :param deadline: The deadline for the reporting period. Must be a datetime object.
+        :param email: The confirmation email. Must be a valid email string.
+        :raises TypeError: If fund_name, reporting_period, deadline, or confirmation_email are not of their respective
+            types.
+        :raises ValueError: If confirmation_email is not a valid email.
+        """
+        if not isinstance(fund_name, str):
+            raise TypeError("Fund name must be a string")
+        if not isinstance(reporting_period, str):
+            raise TypeError("Reporting period must be a string")
+        if not isinstance(reporting_round, int):
+            raise TypeError("Reporting round must be an int")
+        if not isinstance(deadline, datetime.date):
+            raise TypeError("Deadline must be a datetime.date")
+        if not isinstance(email, str):
+            raise TypeError("Deadline must be a str")
+        if not re.match(r"^[A-Za-z0-9._%-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$", email, re.IGNORECASE):
+            raise ValueError("Confirmation email must be a valid email")
+
+        self.fund_name = fund_name
+        self.reporting_period = reporting_period
+        self.reporting_round = reporting_round
+        self.deadline = deadline
+        self.email = email
+
+
+_TF_FUND_CONFIG = FundConfig(
+    fund_name=SupportedFund.TOWNS_FUND,
+    reporting_period="April to September 2023",
+    reporting_round=4,
```

Comment:
Will we have to manually adjust the ```reporting_round``` here everytime a new round begins?

Reply:
Yep, the reporting round, period and deadline. 

Just had a thought that we may want an `active` flag here as well that if true would block users from accessing - otherwise the current option is to delete the config but this doesn't seem as useful

Reply:
Maybe make this clear with some comments in the code, or in a module-level docstring? There's a possibility it doesn't get manually updated and remains out of the date for e.g. round 5. 

Unless we have something like "submission_period" that prevents a user submitting outside of that period, but when I'm not sure we always know what that is? The only problem with an active flag is that somebody might forget to turn it off as it will presumably also be manual.

Reply:
I've renamed the attributes that need changing between rounds to be prefixed with "current". I think it should be clear that these will need modifying between rounds, and it would be immediately apparent in testing if they hadn't been updated.